NServiceBus page on Azure Docs (#1)
kbaley/azure-docs: articles/service-bus-messaging/integrate-with-nservicebus.md

> ---

Author: Please make sure you've read the PR description completely to have an idea of specific things we're looking for feedback on. (This comment isn't for you specifically, I know you're diligent and always read things thoroughly. It's for everyone else.)

> title: Handle complex solutions with NServiceBus

Reviewer: Is the solution that is complex or the problem?

Reviewer 2: Either or both. The fact remains that having raw ASB in a solution makes it even more complex.

> title: Handle complex solutions with NServiceBus

Reviewer 3: Not a fan of "handle" or "solution" as both terms seem vague. The description is great. Why not just boil it down and use it as the title? Ex: Build complex systems with NServiceBus.

Reviewer 4: I'm not sure it's a good idea to build in an assumption that solutions built with NServiceBus are complex. The ultimate goal is to solve complex _problems_ in the simplest way.

Maybe:

```suggestion
title: Solve complex problems with NServiceBus
```

Note that, although I'm making this suggestion here, if it (or something similar) is used, this change should ripple through the rest of the document, i.e. the description, section titles, and the text itself.

Reviewer 2: > Solve complex problems with NServiceBus

Adam, I like the solving part, but not the generalized "problems". Keep in mind, we get a single shot, a single page, to talk about NServiceBus.

Reviewer 4: @SeanFeldman can you suggest something else? "solutions" is also general, and I don't think we should talk about "complex solutions". Rather "complex problems".

Author: I like Britt's suggestion of "build": Build complex systems with NServiceBus

Reviewer 4: Playing devil's advocate: "Build complex systems with NServiceBus. Build simple solutions with {insert rival product here}.".

It's the _problems_ that are complex. The complexity of the solution is a function of many variables, and ideally, is low. I.e. the best solution to complex problems are simple.

Author: Thanks for the suggestions. We're going with a variation on @adamralph's suggestion:

> Solve complex problems with NServiceBus and Service Bus

> description: Learn how to develop complex distributed systems on Service Bus using the NServiceBus framework.

Reviewer 4: ```suggestion
description: Learn how to solve complex problems with distributed systems on Service Bus using the NServiceBus framework.
```

author: kbaley, SeanFeldman                      
ms.author: spelluru                  
ms.service: service-bus-messaging    
ms.topic: how-to                     
ms.date: 06/01/2021                  
ms.custom: template-how-to           
---

<!--
Remove all the comments in this template before you submit a PR.
-->

<!-- This is included for the sake of the RFC. Refer to https://docs.microsoft.com/en-us/azure/service-bus-messaging/ to see the options available for the table of contents -->
<!-- Note that more than one option MAY be available but we haven't verified with Microsoft -->

Possible table of contents locations:

- Tutorials
  - Simplify Service Bus with NServiceBus (link to this article)

- How to guides
  - Develop

>     - Handle complex solutions with NServiceBus (link to this article)

Reviewer: I prefer `How to guides` over `Tutorials`, but it's not a strong preference.

Author: For visibility sake, we're going with Tutorials because it's higher up in the hierarchy and has only two other articles in it vs. "How to guides" which is a lot more overcrowded.

Microsoft will have the final say on this, of course, but we'll start here.

- Reference
  - .NET

>     - Handle complex solutions with ServiceBus (link to this article)

Reviewer: 👎🏻 - `Reference` doesn't sound right to me

Author: It may not have been clear: the suggested locations already exist in Microsoft's TOC. And we have the option of at least suggesting that we put links in more than one location in the TOC.

Reviewer: No, it was crystal clear. I don't think we should list such an article in the Reference or Resources locations.

Reviewer 2: @mauroservienti, so you are saying it's not a good idea to link to the article from the only location that specifically calls out for .NET and .NET developers would be looking for?
![image](https://user-images.githubusercontent.com/1309622/121213753-c2195d80-c83b-11eb-9eb6-c04f0cc9e879.png)

Author: As Sean mentioned, one reason to include it here is for discoverability. We'd tried to think as a .NET developer looking through the table of contents.

Reviewer: My reasoning is based on the way I'd look for such an article. I would never look into a reference section. I'd expect a reference section to contain only API documentation

Author: Agreed. Linking to the article under references isn't a good idea. We've decided to include a link directly to the ASB transport page in docs since that's more in line with the other articles in this section.

- Resources

>   - NServiceBus (link to our docs or marketing site)

Reviewer: 👎🏻 - `Resources` doesn't sound right to me

Reviewer 2: I'm just going to visualize what else is in that category and ask why this doesn't sound right to you @mauroservienti? Thank you.

![image](https://user-images.githubusercontent.com/1309622/121214016-ff7deb00-c83b-11eb-900f-2390ea61aa32.png)

Reviewer: Looking at the list content my feeling is that an article titled "Handle complex solutions with NServiceBus" is not a good fit. But, if we're looking for reach then maybe the more places the better.

Author: For the Resources section, we would link directly to our marketing (or docs) site. Not to the article we're writing.

<!-- 1. H1
Required. Start your H1 with a verb. Pick an H1 that clearly conveys the task the 
user will complete.
-->
# Build complex solutions with NServiceBus

<!-- 2. Introductory paragraph 
Required. Lead with a light intro that describes, in customer-friendly language, 
what the customer will learn, or do, or accomplish. Answer the fundamental “why 
would I want to do this?” question. Keep it short.
-->

> NServiceBus is a messaging framework built on top of Service Bus to abstract infrastructure concerns so that developers can focus on business logic. In this guide, we'll build a solution to exchange messages between two services. We'll also show how to automatically retry failing messages and review options for hosting these services in Azure.

Reviewer 3: ```suggestion
NServiceBus is a messaging framework built on top of Service Bus that helps developers focus on business logic by abstracting infrastructure concerns. In this guide, we'll build a solution to exchange messages between two services. We'll also show how to automatically retry failing messages and review options for hosting these services in Azure.
```
